hpx/async.hpp

```diff
@@ -13,7 +13,7 @@
 #include <hpx/lcos/async_continue.hpp>
 #include <hpx/lcos/future.hpp>
 #include <hpx/runtime/launch_policy.hpp>
-#include <hpx/runtime/threads/thread.hpp>
+#include <hpx/thread.hpp>
```

Comment:
Can we use `threads` as the name of this module, please? This would avoid naming problems compared to `hpx::thread`.

Reply:
Possible `threading` would work as well.

Reply:
Yeah, let's do that. I guess `threading` is more consistent with `threading_base`.

Reply:
I've renamed it `threading`.